FWCore/FWLite/interface/BranchMapReader.h

```diff
@@ -86,22 +87,26 @@ namespace fwlite {
     int getFileVersion(TFile* file);
     int getFileVersion() const { return  fileVersion_;}
 
-    TFile* getFile() const { return strategy_->currentFile_; }
-    TTree* getEventTree() const { return strategy_->eventTree_; }
-    TTree* getLuminosityBlockTree() const { return strategy_->luminosityBlockTree_; }
-    TTree* getRunTree() const { return strategy_->runTree_; }
+    TFile const* getFile() const { return strategy_->currentFile_; }
+    TFile* getFile() { return strategy_->currentFile_; }
+    TTree const* getEventTree() const { return strategy_->eventTree_; }
+    TTree* getEventTree() { return strategy_->eventTree_; }
+    TTree const* getLuminosityBlockTree() const { return strategy_->luminosityBlockTree_; }
+    TTree* getLuminosityBlockTree() { return strategy_->luminosityBlockTree_; }
+    TTree const* getRunTree() const { return strategy_->runTree_; }
+    TTree* getRunTree() { return strategy_->runTree_; }
     TUUID getFileUUID() const { return strategy_->fileUUID_; }
     Long_t getEventEntry() const { return strategy_->eventEntry_; }
     Long_t getLuminosityBlockEntry() const { return strategy_->luminosityBlockEntry_; }
     Long_t getRunEntry() const { return strategy_->runEntry_; }
     const std::vector<edm::BranchDescription>& getBranchDescriptions();
-    const edm::BranchListIndexes& branchListIndexes() const { strategy_->updateMap(); return strategy_->branchListIndexes(); }
+    const edm::BranchListIndexes& branchListIndexes() { strategy_->updateMap(); return strategy_->branchListIndexes(); }
```

Comment:
I'm guessing this is intended to be a caching strategy so the function is meant to be semantically `const`